pkg/webhook/resources/upgrade/validator.go

```diff
@@ -514,3 +524,39 @@ func (v *upgradeValidator) getKubeletStatsSummary(nodeName, kubeletURL string) (
 	}
 	return summary, nil
 }
+
+func (v *upgradeValidator) checkCerts(version *v1beta1.Version) error {
+	kubernetesIP, err := util.GetKubernetesIP(v.serviceCache)
+	if err != nil {
+		return werror.NewInternalError(fmt.Sprintf("can't get kubernetes ip, err: %+v", err))
+	}
+	if kubernetesIP == "" {
+		err = fmt.Errorf("cluster ip is empty")
+		logrus.WithFields(logrus.Fields{
+			"service.namespace": metav1.NamespaceDefault,
+			"service.name":      "kubernetes",
+		}).WithError(err).Error("cluster ip is empty in the service")
+		return werror.NewInternalError(fmt.Sprintf("can't get kubernetes ip, err: %+v", err))
+	}
+
+	earliestExpiringCert, err := util.GetEarliestExpiringCert(fmt.Sprintf("%s:443", kubernetesIP))
```

Comment:
The code checks certs by using the service's cluster IP. If there are multiple api services behind the services, are they all checked together?

There are a lot of rotatable services in the [doc](https://docs.rke2.io/security/certificates#rotating-client-and-server-certificates-manually). Does this check cover all the services in the doc? 
> The following certificates can be rotated:
admin, api-server, controller-manager, scheduler, rke2-controller, rke2-server, cloud-controller, etcd, auth-proxy, kubelet, kube-proxy.